src/main/java/org/folio/consortia/service/impl/ConsortiumServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package org.folio.consortia.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.consortia.domain.dto.Consortium;
+import org.folio.consortia.domain.entity.ConsortiumEntity;
+import org.folio.consortia.domain.repository.ConsortiumRepository;
+import org.folio.consortia.exception.ResourceAlreadyExistException;
+import org.folio.consortia.service.ConsortiumService;
+import org.springframework.core.convert.ConversionService;
+import org.springframework.stereotype.Service;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class ConsortiumServiceImpl implements ConsortiumService {
+  private final ConsortiumRepository repository;
+  private final ConversionService converter;
+
+  @Override
+  public Consortium save(Consortium consortiumDto) {
+    checkConsortiumNotExistsOrThrow();
+    ConsortiumEntity entity = new ConsortiumEntity();
+    entity.setId(consortiumDto.getId());
+    entity.setName(consortiumDto.getName());
+    ConsortiumEntity consortiumEntity = repository.save(entity);
+    return converter.convert(consortiumEntity, Consortium.class);
+  }
+
+  private void checkConsortiumNotExistsOrThrow() {
+    if (!repository.findAll().isEmpty()) {
```

Comment:
It is not good practive to retieve all records from database to check if it is empty, can you check if some count method exists in reposirory?